src/components/tl/cuda/alltoallv/alltoallv_ce.c

```diff
@@ -0,0 +1,389 @@
+            status = ucc_ee_executor_task_test(task->alltoallv_ce.exec_task[i]);
+            if (status != UCC_OK) {
+                if (status == UCC_OPERATION_INITIALIZED) {
+                    status = UCC_INPROGRESS;
+                }
+                task->super.status = status;
+                return;
+            }
+        }
+        status =
+            ucc_tl_cuda_shm_barrier_start(UCC_TL_TEAM_RANK(team), task->bar);
+        if (ucc_unlikely(status != UCC_OK)) {
+            task->super.status = status;
+            return;
+        }
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_BAR;
+    default:
+        ucc_assert(task->alltoallv_ce.stage == ALLTOALL_CE_STAGE_BAR);
+        break;
+    }
+    task->super.status =
+        ucc_tl_cuda_shm_barrier_test(UCC_TL_TEAM_RANK(team), task->bar);
+    if (task->super.status == UCC_OK) {
+        ucc_tl_cuda_put_sync(task);
+    }
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+
+    if (task->alltoallv_ce.stage != ALLTOALL_CE_STAGE_POST_COPIES) {
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_SYNC;
+    }
+
+    return ucc_progress_queue_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
+}
+
+ucc_status_t
+ucc_tl_cuda_alltoallv_ce_triggered_post_setup(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_status_t        status;
+
+    do {
+        status = ucc_tl_cuda_get_sync(task);
+    } while (status == UCC_INPROGRESS);
+    if (ucc_unlikely(status != UCC_OK)) {
+        return status;
+    }
+
+    status = ucc_tl_cuda_alltoallv_setup_start(task);
+    if (ucc_unlikely(status != UCC_OK)) {
+        ucc_tl_cuda_put_sync(task);
+        return status;
+    }
+
+    do {
+        status = ucc_tl_cuda_alltoallv_setup_test(task);
+    } while (status == UCC_INPROGRESS);
+    if (ucc_unlikely(status != UCC_OK)) {
+        ucc_tl_cuda_put_sync(task);
+        return status;
+    }
+    task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_POST_COPIES;
+
+    return UCC_OK;
+}
+
+//NOLINTNEXTLINE
+size_t ucc_tl_cuda_alltoallv_get_size(const ucc_tl_cuda_task_t *task,
+                                      size_t *sizes, ucc_rank_t block)
+{
+    return sizes[block];
+}
+
+size_t ucc_tl_cuda_alltoallv_get_offset(const ucc_tl_cuda_task_t *task,
+                                        size_t *displ, ucc_rank_t block)
+{
+    return displ[block];
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_init(ucc_tl_cuda_task_t *task)
+{
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+    ucc_coll_args_t    *args = &TASK_ARGS(task);
+    ucc_status_t        status;
+    size_t              data_len;
+
+    if (!UCC_COLL_ARGS_CONTIG_BUFFER(args)) {
+        tl_error(UCC_TL_TEAM_LIB(team), "Do not support non-contiguous buffer");
```

Comment:
I think it's better to use tl_debug here since we will fallback to other TL. I see a lot of errors in internal tests log but test passes anyway
```
[1649680880.572873]     alltoallv_ce.c:344  TL_CUDA ERROR Do not support non-contiguous buffer
```

Reply:
I'm concerned that users might make wrong assumptions if we fall back silently. Perhaps we can just print the error the first time?

Reply:
@kingchc the behavior we are trying to keep in UCC is: be functional as much as possible (means, if smth is not available/not implement try to fallback to other components) and be "silent" by default unless fatal error (like NO_MEM). User can always check with proper LOG_LEVEL setting